Resources/MessageQueue/StompMQConnector.py

```diff
@@ -181,13 +190,20 @@ def subscribe(self, parameters=None):
     if not callback:
       self.log.error("No callback specified!")
 
+    fail = False
     for connection in self.connections.itervalues():
-      listener = StompListener(callback, acknowledgement, connection, mId)
-      connection.set_listener('', listener)
-      connection.subscribe(destination=dest,
-                           id=mId,
-                           ack=ack,
-                           headers=headers)
+      try:
+        listener = StompListener(callback, acknowledgement, connection, mId)
+        connection.set_listener('', listener)
+        connection.subscribe(destination=dest,
+                             id=mId,
+                             ack=ack,
+                             headers=headers)
+      except Exception as e:
+        self.log.error('Failed to subscribe: %s' % e)
+        fail = True
+    if fail:
+      return S_ERROR(EMQUKN, 'Failed to subscribe to at least one broker')
```

Comment:
why not returning in the exception directly ?

Reply:
I am not sure. I think the argument is the following:  this for loop is over all connected brokers (we can have more than one for the same connection), we are tryign to close connection to all brokers. If the error occurs for a given broker should we stop immediatly or try to continue anyway with another brokers.

Reply:
I did not add this logic, it was like that I just cleaned the code a bit.